dashboard/utils_test.py

```diff
@@ -363,9 +367,108 @@ def test_is_enrolled_mmtrack_fa(self):
+    @patch('dashboard.utils.MMTrack.extract_final_grade', autospec=True)
+    def test_has_passed_course(self, extr_grade_mock):
+        """
+        Test for has_passed_course method.
+        V1 new behavior soon to be default
+        """
+        course_id = "course-v1:odl+FOO101+CR-FALL15"
+        grade_mock = MagicMock(passed=True)
+        extr_grade_mock.return_value = grade_mock
+
+        mmtrack = MMTrack(
+            user=self.user,
+            program=self.program,
+            edx_user_data=self.cached_edx_user_data
+        )
+
+        assert mmtrack.has_passed_course(course_id) is True
+        extr_grade_mock.assert_called_once_with(mmtrack, course_id)
+
+        extr_grade_mock.reset_mock()
+        assert extr_grade_mock.call_count == 0
+
+        extr_grade_mock.side_effect = FinalGrade.DoesNotExist
+        with self.assertRaises(FinalGrade.DoesNotExist):
+            mmtrack.has_passed_course(course_id)
+        extr_grade_mock.assert_called_once_with(mmtrack, course_id)
+
+    @override_settings(FEATURES={"FINAL_GRADE_ALGO": "v1"})
+    @patch('dashboard.utils.MMTrack.extract_final_grade', autospec=True)
+    def test_get_final_grade(self, extr_grade_mock):
+        """
+        Test for has_passed_course method.
+        V1 new behavior soon to be default
+        """
+        course_id = "course-v1:odl+FOO101+CR-FALL15"
+        grade_mock = MagicMock(grade=0.57)
+        extr_grade_mock.return_value = grade_mock
+
+        mmtrack = MMTrack(
+            user=self.user,
+            program=self.program,
+            edx_user_data=self.cached_edx_user_data
+        )
+
+        assert round(mmtrack.get_final_grade(course_id)) == 57.0
+        extr_grade_mock.assert_called_once_with(mmtrack, course_id)
+
+        extr_grade_mock.reset_mock()
+        assert extr_grade_mock.call_count == 0
+
+        extr_grade_mock.side_effect = FinalGrade.DoesNotExist
+        with self.assertRaises(FinalGrade.DoesNotExist):
+            mmtrack.has_passed_course(course_id)
+        extr_grade_mock.assert_called_once_with(mmtrack, course_id)
+
+    @patch('grades.models.FinalGrade.objects.get', autospec=True)
+    def test_extract_final_grade(self, get_grades_mock):
+        """
+        Test for extract_final_grade method.
+        """
+        course_id = "course-v1:odl+FOO101+CR-FALL15"
+        grade_mock = MagicMock(grade=0.97, passed=True)
+        get_grades_mock.return_value = grade_mock
+
+        mmtrack = MMTrack(
+            user=self.user,
+            program=self.program,
+            edx_user_data=self.cached_edx_user_data
+        )
+
+        res = mmtrack.extract_final_grade(course_id)
+        assert res.grade == grade_mock.grade
+        assert res.passed == grade_mock.passed
+        get_grades_mock.assert_called_once_with(
+            course_run__edx_course_key=course_id,
+            status=FinalGradeStatus.COMPLETE,
+            user=self.user
+        )
+        get_grades_mock.reset_mock()
```

Comment:
I left the following assertion because I want to test that there is only one call to the database, so I need to make 2 calls anyway, but I put the test with the exception in another test